Mathlib/Tactic/Simproc/Matrix.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import Mathlib.Data.Matrix.Reflection
+
+/-!
+# Simprocs for matrices
+
+In this file we provide a simproc for transpose of explicit matrices.
+-/
+
+open Matrix
+open Lean Meta Simp Simproc Tactic Qq
+
+attribute [-simp] cons_transpose
+
+namespace Qq
+
+/-- Decompose a vector expression into a vector of expressions. -/
+def piFinLit! {u : Level} {n : ℕ} {R : Q(Type u)} (e : Q(Fin $n → $R)) :
```

Comment:
What's the `!` for here? Usually it indicates a panic, but since this returns an `Option` likely `?` would be more appropriate.

You might find you can reuse some code from the `Matrix.cons_val` simproc; or at least, you should add a comment explaining why doing so isn't worth it.

Reply:
ah, I might have entirely misunderstood the `!` and `?` convention. I used `?` to refer to expressions and `!` to refer to "real" objects.

Reply:
Yeah, that's `!` because it panics if you call it on `q("a string literal")`

Reply:
done